### ISSUE TITLE
irqs_disable() possible out-of-boundary table access.

### ISSUE THREAD

**Issue by sberbz** [nutanix/libvfio-user#790]
While fuzzing SPDK's vfio-user transport, encounter an issue with function `irqs_disable()` that seems to be out-of-boundary table access.

Here is a comment describing the issue: 
https://github.com/spdk/spdk/issues/3289#issuecomment-1976712750

Here is another instance of failure: 
https://10.211.11.56/job/long-fuzz-vfio-phy-autotest/97/consoleText

**Comment by jlevon** [comment]
```
315     // Ensure irq_set's start and count are within bounds.                       
316     if ((irq_set->start >= vfu_ctx->irq_count[irq_set->index]) ||                
317         (irq_set->start + irq_set->count > vfu_ctx->irq_count[irq_set->index])) {
318         line = __LINE__;                                                         
319         goto invalid;                                                            
320     }                                                                            
```

**Comment by jlevon** [comment]
fixed by #791
